src/Qir/Tools/Executable/QirExecutableRunner.cs

```diff
@@ -37,6 +38,11 @@ public async Task RunExecutableAsync(FileInfo executableFile, Stream stream, str
             using var streamWriter = new StreamWriter(stream);
             await streamWriter.WriteAsync(output);
             logger?.LogInformation($"Executable has finished running. Result code: {process.ExitCode}");
+
+            if (process.ExitCode != 0)
```

Comment:
The common pattern is that success means exist code = 0. @cesarzc let me know if that's not the case for the simulator executable.

Reply:
Exit code 0 means success in the simulator executable.